File: src/main/java/com/github/kindrat/cassandra/client/filter/fields/LikePredicate.java
package com.github.kindrat.cassandra.client.filter.fields;

import com.github.kindrat.cassandra.client.ui.DataObject;

import java.util.regex.Pattern;

import static java.util.Objects.nonNull;

public class LikePredicate extends RowPredicate {
    public LikePredicate(String field, String value) {
        super(field, value);
    }

    @Override
    public boolean test(DataObject data) {
        String value = getValue();
        Pattern pattern = Pattern.compile(value);
        Object actualValue = data.get(getField());
        //noinspection unchecked
        return nonNull(actualValue)
                && pattern.matcher(actualValue.toString()).find();
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/fields/LikePredicate.kt
package com.github.kindrat.cassandra.client.filter.fields

import com.github.kindrat.cassandra.client.ui.DataObject
import java.util.regex.Pattern

class LikePredicate(field: String, value: String) : RowPredicate(field, value) {

    override fun test(data: DataObject): Boolean {
        val pattern = Pattern.compile(value)
        val actualValue = data[field]
        return pattern.matcher(actualValue.toString()).find()
    }
}